### PR TITLE
Add integrator friendly docker compose

### PR DESCRIPTION
**Pull request by malexandru-rdx**
This PR adds:

1.  `node.key.create_if_missing` config option which when enabled, if no node key is provided, a new one is created on the fly - this is needed to remove the step of creating a private key when wanting to quickly spin up a fullnode. Note: Hardcoded node key would not work since the protocol does not guarantee all IPs (or which IP) under same node key
2. Add `testnet-node` - docker-compose file to connect to RCv1 testnet for integrators  
3. Remove unused `docker/Dockerfile.build` and add multi-stage `Dockerfile`

### CODE REVIEW

File: .vscode/settings.json
Comment (dhedey) [inline .vscode/settings.json:2]: This likely needs to be explicitly removed - it shouldn't be source controlled given the .gitignore.

File: core/src/main/java/com/radixdlt/keys/PersistedBFTKeyManager.java
Comment (dhedey) [inline core/src/main/java/com/radixdlt/keys/PersistedBFTKeyManager.java:83]: I don't think is ideal - if this node restarts, its key will be lost - which I don't think we want.

Instead, one of the `Keys.readKey` takes a `create` parameter - which creates a key at the given location - we should be using that.

That way the key will be persisted across docker image restarts.

File: docker/Dockerfile.core
Comment (dhedey) [inline docker/Dockerfile.core:47]: I don't think we necessarily need to put this here, but I guess it shouldn't hurt :)

File: testnet-node/docker-compose.yml
Comment (dhedey) [inline testnet-node/docker-compose.yml:14]: Should this be removed? Or if it's an option that developers might want to use, we should add a comment explaining why they might want to change this

File: testnet-node/docker-compose.yml
Comment (dhedey) [inline testnet-node/docker-compose.yml:7]: Might warrant a comment to explain this is coming soon?

File: testnet-node/docker-compose.yml
Comment (dhedey) [inline testnet-node/docker-compose.yml:1]: I imagine we probably want:

* A clear prompt in the main README: "If you're looking to develop an integration to run a Core API node, see the [./testnet-node](testnet-node folder)."
* A clear README in this folder explaining that this is intended for development; how to use this docker file, where to find docs for integrations, and where to find docs for configuring a node in production.

**Review by dhedey — CHANGES_REQUESTED** [review]
So this PR confusingly has an old version of Kim's PR in it.

I think what we should do is:
* Rebase Kim's PR onto `release/rcnet-v1` and merge into `release/rcnet-v1`. 
* Rebase/merge this into `release/rcnet-v1` after Kim's PR has been merged. (for the time being, this PR could have Kim's PR as a base to avoid confusion about what's under review)
* Then merge everything into develop.

File: core/src/main/java/com/radixdlt/keys/PersistedBFTKeyManager.java
Comment (malexandru-rdx) [reply core/src/main/java/com/radixdlt/keys/PersistedBFTKeyManager.java:83]: > I don't think is ideal - if this node restarts, its key will be lost - which I don't think we want.
I am viewing from the perspective where anything using private keys should be intentional and explicitly manage by the user - i.e. you want (or need?) your private key to persist - create and manage it yourself :). I don't want people to rely on a private key generated inside the docker.

This is a mere convenience functionality (I don't really care about details, just give me a quick way to scoop in the network).

File: testnet-node/README.md
Comment (dhedey) [inline testnet-node/README.md:4]: Some of this is really good, but some of it is a little too detailed.

I'd recommend replacing all of the public key big with this:
```
A node in the network is identified by its public key. A keystore containing a keypair is automatically generated on startup and stored inside a docker volume called `key-data`, protected by default with password `"password"`.
```

The ledger stuff is good :)

File: testnet-node/README.md
Comment (dhedey) [inline testnet-node/README.md:15]: What version of docker was this tested on?

We say this in the development readme 👍 
```
- Docker version 20.10+
- `docker-compose` version 1.25+
```

File: testnet-node/README.md
Comment (dhedey) [reply testnet-node/README.md:4]: I'd also suggest mentioning:
* `The node connects to the latest public testnet (as of April/May 2023, this is RCnet V1, known by its logical name kisharnet).`
* `[Documentation for integrators is available here](https://docs.google.com/document/d/1cjc7_alyzIb2QQIGGn1PEpJyjrMRZYHq3VwkOXRP8J0/edit#).`

File: testnet-node/README.md
Comment (dhedey) [inline testnet-node/README.md:18]: Nice 👍

File: README.md
Comment (dhedey) [inline README.md:22]: I'd suggest to make it really easy to find, we should have an ## Integrators section even above `## SubDirectories` - and say - if you're looking to run a node for developing an integration, see `[details on running a testnet node for development](./testnet-node)` and the `[Babylon documentation for exchanges and integrators](https://docs.google.com/document/d/1cjc7_alyzIb2QQIGGn1PEpJyjrMRZYHq3VwkOXRP8J0/edit#)`

**Review by dhedey — APPROVED** [review]
Really nice. A few final comments 👍

**Pull request merged**
